### PR TITLE
[Misc] Add interruptible execution

### PR DESCRIPTION
**Pull request by ibmibmibm**
* Support asynchronous execution
* Add new option for timeout execution
* Add new option for support interruptible execution in AOT

### CODE REVIEW

File: include/vm/async.h [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+  Async &operator=(Async &&Other) noexcept {
+    swap(*this, Other);
+    return *this;
+  }
+
+  bool valid() const noexcept { return Future.valid(); }
+
+  T get() { return Future.get(); }
+
+  void wait() const { return Future.wait(); }
```
Comment (hydai) [inline include/vm/async.h:51]: Why do we need to return a void function?

File: include/api/wasmedge/wasmedge.h [hunk 2]
```diff
@@ -2660,6 +2725,96 @@ WASMEDGE_CAPI_EXPORT extern WasmEdge_Result WasmEdge_VMRunWasmFromASTModule(
...
+/// This is the first step to invoke a WASM function step by step.
+/// Load and parse the WASM module from the file path. You can then call
+/// `WasmEdge_VMValidate` for the next step.
+///
+/// \param Cxt the WasmEdge_VMContext.
+/// \param Path the NULL-terminated C string of the WASM file path.
+///
+/// \returns WasmEdge_Result. Call `WasmEdge_ResultGetMessage` for the error
+/// message.
+
```
Comment (hydai) [inline include/api/wasmedge/wasmedge.h:2817]: I am not sure where does this section belong to?

**Review by hydai — CHANGES_REQUESTED** [review]

File: include/vm/async.h (hunk 1 above)
Comment (ibmibmibm) [reply include/vm/async.h]: Done.

File: include/api/wasmedge/wasmedge.h (hunk 2 above)
Comment (ibmibmibm) [reply include/api/wasmedge/wasmedge.h]: Removed.

**Review by hydai — DISMISSED** [review]

**Review by hydai — APPROVED** [review]

**Review by q82419 — APPROVED** [review]

**Pull request merged**